app/controllers/errors_controller.rb

```diff
@@ -1,7 +1,15 @@
 class ErrorsController < ApplicationController
+  skip_before_action :authenticate_user!
   skip_before_action :verify_authenticity_token
   skip_before_action :check_user_access
 
+  before_action do
+    # soft authenticate so we can serve error pages to users who aren't logged
+    # in - the hard `authenticate_user!` approach is incompatible with a rails
+    # error app
```

Comment:
Minor: would be helpful to say this just retrieves any available session, instead of forcing a login.